cmd/kubeadm/app/phases/certs/pkiutil/pki_helpers.go

```diff
@@ -63,26 +63,52 @@ func NewCertAndKey(caCert *x509.Certificate, caKey *rsa.PrivateKey, config certu
 }
 
 func WriteCertAndKey(pkiPath string, name string, cert *x509.Certificate, key *rsa.PrivateKey) error {
-	certificatePath, privateKeyPath := pathsForCertAndKey(pkiPath, name)
-
 	if cert == nil {
 		return fmt.Errorf("certificate cannot be nil when writing to file")
 	}
-	if cert == nil {
+	if key == nil {
```

Comment:
since `if (cert|key) == nil` is checked in the WriteKey and WriteCert functions, you can omit these checks